### PR TITLE
ci: (windows) misc

### PR DESCRIPTION
**Pull request by jmoguillansky-gpsw**
Split mingw and msvc tests into separate workflows
Improve vcpkg caching mechanism

### CODE REVIEW

**Comment by cboesch-gpsw** [comment]
Isn't this going to break the badge in the README?

File: .github/workflows/ci_msvc.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+          pip.exe install meson
+          cd /mnt/c/vcpkg
+          ./vcpkg.exe install pthreads:x64-windows opengl-registry:x64-windows ffmpeg[ffmpeg,ffprobe]:x64-windows sdl2:x64-windows
+          # TODO: remove vcpkg integrate install
+          ./vcpkg.exe integrate install
+
+      - name: Build
+        run: |
+          # Currently, Meson has a bug where the include paths need to be on the same drive as the project directory
+          # see https://github.com/mesonbuild/meson/issues/8329
```
Comment (cboesch-gpsw) [inline .github/workflows/ci_msvc.yml:48]: please do not slip in changes in a code move: preserve the content of the rules exactly the same. These changes must be discussed separately

Comment (jmoguillansky-gpsw) [reply .github/workflows/ci_msvc.yml:48]: I'm using the built-in vcpkg but it's on C drive.  The project is on D drive.  There's a bug in meson where the dependencies need to be on the same drive.

Comment (cboesch-gpsw) [reply .github/workflows/ci_msvc.yml:48]: That's not what my remark is about: you're hiding functional changes in a move commit

Comment (jmoguillansky-gpsw) [reply .github/workflows/ci_msvc.yml:48]: it's not only a move commit, see the commit message.  But I will split it up into atomic commits like you prefer

Comment (jbelaston-gpsw) [reply .github/workflows/ci_msvc.yml:48]: +1
Yeah I think it's really difficult here to do the review of a code which has been moved and updated... :disappointed: 
So, please, here my proposal :slightly_smiling_face: :
- one commit only for the move (without any other change)
- then another one for the change

**Review by jbelaston-gpsw — CHANGES_REQUESTED** [review]

**Comment by jmoguillansky-gpsw** [comment]
closed, already merged in a different PR

**Pull request closed without merging**